packages/starknet-snap/src/state/transaction-state-manager.ts

```diff
@@ -0,0 +1,305 @@
+import type { constants, TransactionType } from 'starknet';
+import {
+  TransactionFinalityStatus,
+  TransactionExecutionStatus,
+} from 'starknet';
+import { assert, enums, number } from 'superstruct';
+
+import type { Transaction, SnapState } from '../types/snapState';
+import { TransactionStatusType } from '../types/snapState';
+import type { IFilter } from './filter';
+import {
+  BigIntFilter,
+  ChainIdFilter as BaseChainIdFilter,
+  StringFllter,
+  Filter,
+} from './filter';
+import { StateManager, StateManagerError } from './state-manager';
+
+export type ITxFilter = IFilter<Transaction>;
+
+export class ChainIdFilter
+  extends BaseChainIdFilter<Transaction>
+  implements ITxFilter {}
+
+export class ContractAddressFilter
+  extends BigIntFilter<Transaction>
+  implements ITxFilter
+{
+  dataKey = 'contractAddress';
+}
+export class SenderAddressFilter
+  extends BigIntFilter<Transaction>
+  implements ITxFilter
+{
+  dataKey = 'senderAddress';
```

Comment:
shall we call senderAddress simply address ? As we don't always send tokens, sometimes we execute for instance.

Reply:
thats include to change the existing txn state from the snap, as well as the type

i would suggest do it in other PR, as this PR is to have mgr to fetch the data from current state 

but i will add a remark on our snapState type